src/main/java/hexgui/util/StringUtils.java

```diff
@@ -0,0 +1,245 @@
+    return buffer.toString();
+  }
+
+  /** Check if string is null, empty, or contains only whitespaces. */
+  public static boolean isEmpty(String s) {
+    if (s == null) return true;
+    for (int i = 0; i < s.length(); ++i) if (!Character.isWhitespace(s.charAt(i))) return false;
+    return true;
+  }
+
+  /** Converts all whitespace characters to a single ' '. */
+  public static String cleanWhiteSpace(String str) {
+    StringReader reader = new StringReader(str);
+    StringBuilder ret = new StringBuilder();
+
+    boolean white = false;
+    while (true) {
+      int c;
+      try {
+        c = reader.read();
+      } catch (Throwable t) {
+        System.out.println("Something bad happened!");
+        break;
+      }
+
+      if (c == -1) break;
+      if (c == ' ' || c == '\n' || c == '\t') {
+        if (!white) ret.append(" ");
+        white = true;
+      } else {
+        white = false;
+        ret.append((char) c);
+      }
+    }
+    return ret.toString();
+  }
+
+  public static Vector<HexPoint> parsePointList(String str, String sep) {
+    Vector<HexPoint> ret = new Vector<HexPoint>();
+    String cleaned = cleanWhiteSpace(str.trim());
+    if (cleaned.length() == 0) return ret;
+
+    String[] pts = cleaned.split(sep);
+    for (int i = 0; i < pts.length; i++) {
+      HexPoint p = HexPoint.get(pts[i].trim());
+      ret.add(p);
+    }
+    return ret;
+  }
+
+  public static Vector<Pair<HexColor, HexPoint>> parseVariation(String str) {
+    Vector<Pair<HexColor, HexPoint>> ret = new Vector<Pair<HexColor, HexPoint>>();
+
+    Vector<Pair<String, String>> pairs = StringUtils.parseStringPairList(str.trim());
+    for (int i = 0; i < pairs.size(); ++i) {
+      HexColor color = (pairs.get(i).first.charAt(0) == 'B') ? HexColor.BLACK : HexColor.WHITE;
+      HexPoint point = HexPoint.get(pairs.get(i).second);
+      ret.add(new Pair<HexColor, HexPoint>(color, point));
+    }
+    return ret;
+  }
+
+  public static Vector<HexPoint> parsePointList(String str) {
+    return parsePointList(str, " ");
+  }
+
+  public static Vector<String> parseStringList(String str) {
+    Vector<String> ret = new Vector<String>();
+    String cleaned = cleanWhiteSpace(str.trim());
+    if (cleaned.length() == 0) return ret;
+
+    String[] strs = cleaned.split(" ");
+    for (int i = 0; i < strs.length; i++) {
+      String cur = strs[i].trim();
+      ret.add(cur);
+    }
+    return ret;
+  }
+
+  public static Vector<Pair<String, String>> parseStringPairList(String str) {
+    Vector<Pair<String, String>> ret = new Vector<Pair<String, String>>();
+    String cleaned = cleanWhiteSpace(str.trim());
+    if (cleaned.length() == 0) return ret;
+
+    String[] strs = cleaned.split(" ");
+    for (int i = 0; i < strs.length; i += 2) {
+      String c1 = strs[i].trim();
+      String c2 = strs[i + 1].trim();
+      ret.add(new Pair<String, String>(c1, c2));
+    }
+    return ret;
+  }
+
+  public static Vector<VC> parseVCList(String str) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.